ToolBox/FxcmDownloader/FxcmDataDownloader.cs

```diff
@@ -0,0 +1,422 @@
+        private readonly List<TradeBar> _currentBars = new List<TradeBar>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FxcmDataDownloader"/> class
+        /// </summary>
+        public FxcmDataDownloader(string server, string terminal, string userName, string password)
+        {
+            _server = server;
+            _terminal = terminal;
+            _userName = userName;
+            _password = password;
+
+            LoadInstruments();
+        }
+
+        /// <summary>
+        /// Loads the instrument list from the instruments.txt file
+        /// </summary>
+        /// <returns></returns>
+        private void LoadInstruments()
+        {
+            if (!File.Exists(InstrumentsFileName))
+                throw new FileNotFoundException(InstrumentsFileName + " file not found.");
+
+            _instruments = new Dictionary<Symbol, LeanInstrument>();
+
+            var lines = File.ReadAllLines(InstrumentsFileName);
+            foreach (var line in lines)
+            {
+                var tokens = line.Split(',');
+                if (tokens.Length >= 3)
+                {
+                    var fxcmSymbol = tokens[0];
+                    var symbol = ConvertFxcmSymbolToLeanSymbol(fxcmSymbol);
+                    _instruments.Add(symbol, new LeanInstrument
+                    {
+                        Symbol = symbol,
+                        Name = tokens[1],
+                        Type = (SecurityType)Enum.Parse(typeof(SecurityType), tokens[2])
+                    });
+
+                    _mapInstrumentSymbols[symbol] = fxcmSymbol;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Converts an FXCM symbol to a Lean Symbol instance
+        /// </summary>
+        /// <param name="fxcmSymbol">The FXCM symbol</param>
+        /// <returns>A Lean symbol</returns>
+        private static Symbol ConvertFxcmSymbolToLeanSymbol(string fxcmSymbol)
+        {
+            return new Symbol(fxcmSymbol.Replace("/", "").ToUpper());
+        }
+
+        /// <summary>
+        /// Converts a Java Date value to a UTC DateTime value
+        /// </summary>
+        /// <param name="javaDate">The Java date</param>
+        /// <returns>A UTC DateTime value</returns>
+        private static DateTime FromJavaDateUtc(Date javaDate)
+        {
+            var cal = Calendar.getInstance();
+            cal.setTimeZone(TimeZone.getTimeZone("UTC"));
+            cal.setTime(javaDate);
+
+            // note that the Month component of java.util.Date  
+            // from 0-11 (i.e. Jan == 0)
+            return new DateTime(cal.get(Calendar.YEAR),
+                                cal.get(Calendar.MONTH) + 1,
+                                cal.get(Calendar.DAY_OF_MONTH),
+                                cal.get(Calendar.HOUR_OF_DAY),
+                                cal.get(Calendar.MINUTE),
+                                cal.get(Calendar.SECOND));
```

Comment:
@mchandschuh Thanks, note taken.

Another one here (to be moved and made public): https://github.com/QuantConnect/Lean/blob/master/Brokerages/Fxcm/FxcmBrokerage.Util.cs#L217